server/channels/app/channel_test.go

```diff
@@ -59,7 +64,10 @@ func TestPermanentDeleteChannel(t *testing.T) {
 		CallbackURLs: []string{"http://foo"},
 	})
 	require.Nil(t, err)
-	defer th.App.DeleteOutgoingWebhook(outgoing.Id)
+	defer func() {
+		deferErr := th.App.DeleteOutgoingWebhook(outgoing.Id)
+		require.Nil(t, deferErr)
+	}()
```

Comment:
This is throwing a NPE. Can you please take a look? Thanks!

```
panic: runtime error: invalid memory address or nil pointer dereference
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x25d44a5]

goroutine 57576 [running]:
testing.tRunner.func1.2({0x2aff1c0, 0x535c370})
	/usr/local/go/src/testing/testing.go:1631 +0x24a
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1634 +0x377
panic({0x2aff1c0?, 0x535c370?})
	/usr/local/go/src/runtime/panic.go:770 +0x132
github.com/mattermost/mattermost/server/v8/channels/app.TestPermanentDeleteChannel.func3()
	/mattermost/server/channels/app/channel_test.go:52 +0x25
panic({0x2aff1c0?, 0x535c370?})
	/usr/local/go/src/runtime/panic.go:770 +0x132
github.com/mattermost/mattermost/server/v8/channels/app.TestPermanentDeleteChannel.func4()
	/mattermost/server/channels/app/channel_test.go:68 +0x25
github.com/mattermost/mattermost/server/v8/channels/app.TestPermanentDeleteChannel(0xc004c6ad00)
	/mattermost/server/channels/app/channel_test.go:86 +0x891
testing.tRunner(0xc004c6ad00, 0x3119490)
	/usr/local/go/src/testing/testing.go:1689 +0xfb
created by testing.(*T).Run in goroutine 1
	/usr/local/go/src/testing/testing.go:1742 +0x390
```

Reply:
This was a tricky one.

Introducing the anonymous function for deferring, lead to `outgoing.Id` being null. Because `defer` without anonymous function wrapper puts the call on the stack with the current value of `outgoing.Id` at the moment of the `defer` being defined. Now with the wrapper `outgoing.Id` is evaluated at a later point, so we have to introduce a temp variable to hold the value for us.